src/parser/mod.rs

```diff
@@ -1345,6 +1355,38 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parses method call expression
+    ///
+    /// Returns `Result<{parsed method-call-expr},{orig_expr}>`
+    fn try_parse_method(&mut self, orig_expr: Expr) -> Result<Result<Expr, Expr>, ParserError> {
+        if !self.dialect.supports_methods() {
+            return Ok(Err(orig_expr));
+        }
+        let method_chain = self.maybe_parse(|p| {
```

Comment:
I think we can simplify the return type a bit if we do something like
```rust
let method_chain = self.maybe_parse(|p| {
  let mut method_chain = Vec::new();
  loop {
    // ...
    method_chain.push(func);
  }
  if method_chain.is_empty() {
    p.expected("function")
  } else {
    Ok(method_chain)
  }
})?;

if let Some(method_chain) {
  Ok(Expr::Method{Method { expr, method_chain }})
} else {
  Ok(expr)
}
```
that would remove the nested result type `Result<Result<Expr, Expr>>` -> `Result<Expr>` as usual